tests/components/home_connect/test_diagnostics.py

```diff
@@ -26,10 +30,43 @@ async def test_async_get_config_entry_diagnostics(
     get_appliances: MagicMock,
     snapshot: SnapshotAssertion,
 ) -> None:
-    """Test setup and unload."""
+    """Test config entry diagnostics."""
     get_appliances.side_effect = get_all_appliances
     assert config_entry.state == ConfigEntryState.NOT_LOADED
     assert await integration_setup()
     assert config_entry.state == ConfigEntryState.LOADED
 
     assert await async_get_config_entry_diagnostics(hass, config_entry) == snapshot
+
+
+@pytest.mark.usefixtures("bypass_throttle")
+async def test_device_diagnostics(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    integration_setup: Callable[[], Awaitable[bool]],
+    setup_credentials: None,
+    appliance: MagicMock,
+    get_appliances: MagicMock,
+    device_registry: dr.DeviceRegistry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test config entry diagnostics."""
+    get_appliances.return_value = [appliance]
+    appliance.status.update(
+        HomeConnectAppliance.json2dict(
+            load_json_object_fixture("home_connect/settings.json")
+            .get(appliance.name)
+            .get("data")
+            .get("settings")
+        )
+    )
+    appliance.get_programs_available.return_value = [
+        "Dishcare.Dishwasher.Program.Eco50"
+    ]
```

Comment:
Why is all this needed?
Does it not work directly with `get_appliances.side_effect = get_all_appliances` ?